core/spec.md

```diff
@@ -369,20 +365,133 @@ information about each data type):
 
 This specification defines the following terms:
 
-#### Group
+- **Group**
 
-An entity that acts as a collection of related Resources.
+Groups, as the name implies, is a mechanism by which related Resources are
+arranged together under a single collection - the Group. The reason for the
+grouping is not defined by this specification, so the owners of the Registry
+can choose to define (or enforce) any pattern they wish. In this sense, a
+Group is similar to a "directory" on a filesystem.
 
-#### Registry
+- **Registry**
 
 An implementation of this specification. Typically, the implementation would
 include model-specific Groups, Resources and extension attributes.
 
-#### Resource
+- **Resource**
 
-A Resource is typically the main entity that is stored within a Registry
-Service. A Resource MUST exist within the scope of a Group and it MAY be
-versioned.
+Resources represent the main data of interest for the Registry. In the
+filesystem analogy, these would be the "files". All Resources exist under a
+single Group and, similar to Groups, have a set of Registry metadata.
+However, unlike a Group which only has Registry metadata, each Resource can
+also have a "document" associated with it. For example, a "schema" Resource
+might have a "schema document" as its "document". This specification places no
+restriction on the type of content stored in the Resource's document.
+Additionally, Resources (unlike Groups) MAY be versioned.
+
+- **Version**
+
+A Version is an instance of a Resource that represents a particular point in
+time state of the Resource. Each Version of a Resource has its own set of
+xRegistry metadata and possibly a domain specific document associated with it.
+Each Resource MUST have at least one Version associated with it.
```

Comment:
I think it's worth clarifying here that if versioning isn't important to the client, this one version can remain the only version.

```suggestion
Each Resource MUST have at least one Version associated with it. If versioning is irrelevant for the use case in which the Resource is used, the default Version can be evolved without creating new ones.
```